### ISSUE TITLE
Support preference overrides using query parameters

### ISSUE THREAD

**Issue by Berni11** [zedeus/nitter#186]
On the nitter homepage, it is possible to turn the replacement of Twitter and YouTube links off. But Twitter and YouTube links are still replaced in the rss feeds. Is there a way to disable link replacement for rss feeds?

**Comment by zedeus** [comment]
No, RSS readers can't easily support Nitter's cookie preferences so this is tricky

**Comment by Berni11** [comment]
Maybe by adding url parameters/query string?

**Comment by dardoor** [comment]
I suggest disabling them by default and leaving the redirection to those who want it. There are browser extensions that can redirect to invidio.us, for example.

**Comment by zedeus** [comment]
It will be done with a url parameter. Making all links go to Twitter instead of Nitter would largely defeat the purpose, and only be useful for the minority that for whatever reason prefers using Twitter directly.

**Comment by Deagan** [comment]
I'm a bit confused, does Nitter currently have functionality to append parameters to the end of the URL, for custom preferences?

I have added ?theme=Twitter&hlsPlayback=true, but this seems to only affect the theme?

**Comment by zedeus** [comment]
Only theme is supported, it's a custom check

**Comment by Deagan** [comment]
> Only theme is supported, it's a custom check, though I do want to support the rest + a way to save it (i.e. being able to export and import preferences)

Ah, okay, thank you for clarifying!

I have created the following userscript to apply customer parameters to the URL on page load, if/when the &parameterName'z are implemented!

`// ==UserScript==
// @name        Nitter Preferences 
// @namespace   Violentmonkey Scripts
// @match       https://birdsite.xanny.family/*
// @version     1.0
// ==/UserScript==

// Remove #m when navigating pages to properly apply Nitter Preferences to URLs.
if (location.href.match("#m")) {
	location.href = location.href.replace("#m", "");
}

// Append Nitter Preferences to end of URL, if not present.
const { href } = window.location;
if (href.slice(-5) !== '=true' && href.slice(-6) !== '=false') {
  window.location.replace(href + '?theme=Pleroma&bidiSupport=true&infiniteScroll=true&mp4Playback=true&hlsPlayback=true&muteVideos=true'); // Replace with custom appended Nitter Preferences.
}

//Scan all links on page, then append a Nitter Preferences string to links.
document.querySelectorAll("a").forEach(function (v) {
    if (v.href.indexOf("https://birdsite.xanny.family") === 0) {
        v.href += "?theme=Pleroma&bidiSupport=true&infiniteScroll=true&mp4Playback=true&hlsPlayback=true&muteVideos=true"; // Replace with custom appended Nitter Preferences.
    }
});`

Thank you for Nitter, I love it!

**Comment by MintMain21** [comment]
This should be added to the [ToDo List](https://github.com/acarasimon96/nitter-ft/issues/1).
